playground/miden-wasm/src/lib.rs

```diff
@@ -142,22 +142,226 @@ pub fn prove_program(asm: &str, inputs_frontend: &str) -> Outputs {
+                }
+                return self.print_vm_state();
+            }
+            //DebugCommand::RewindAll
+            3 => {
+                while let Some(new_vm_state) = self.prev_vm_state() {
+                    self.vm_state = new_vm_state;
+                }
+                return self.print_vm_state();
+            }
+            //DebugCommand::Rewind(cycles)
+            4 => {
+                for _cycle in 0..param {
+                    match self.prev_vm_state() {
+                        Some(new_vm_state) => {
+                            self.vm_state = new_vm_state;
+                        }
+                        None => break,
+                    }
+                }
+                return self.print_vm_state();
+            }
+            //DebugCommand::PrintState
+            5 => return self.print_vm_state(),
+            //DebugCommand::PrintStack
+            6 => self.print_stack(),
+            //DebugCommand::PrintStackItem(index)
+            7 => self.print_stack_item(param.try_into().unwrap()),
+            //DebugCommand::PrintMem
+            8 => self.print_memory(),
+            //DebugCommand::PrintMemAddress(address)
+            9 => self.print_memory_entry(param),
+            //DebugCommand::Clock
+            10 => return format!("{}", self.vm_state.clk),
+            _ => return "Invalid command".to_string(),
+        };
+        return ":)".to_string();
+    }
+
+    /// iterates to the next clock cycle.
+    fn next_vm_state(&mut self) -> Option<VmState> {
+        match self.vm_state_iter.next() {
+            Some(next_vm_state_result) => match next_vm_state_result {
+                Ok(vm_state) => Some(vm_state),
+                Err(err) => {
+                    println!("Execution error: {err:?}");
+                    None
+                }
+            },
+            None => {
+                println!("Program execution complete.");
+                None
+            }
+        }
+    }
+
+    /// iterates to the previous clock cycle.
+    fn prev_vm_state(&mut self) -> Option<VmState> {
+        match self.vm_state_iter.next_back() {
+            Some(prev_vm_state_result) => prev_vm_state_result.ok(),
+            None => {
+                println!("At start of program execution.");
+                None
+            }
+        }
+    }
+
+    // ACCESSORS
+    // --------------------------------------------------------------------------------------------
+
+    /// print general VM state information.
+    pub fn print_vm_state(&self) -> String {
+        return format!("{}", self.vm_state);
+    }
+
+    /// print all stack items.
+    pub fn print_stack(&self) -> String {
+        return format!(
+            "{}",
+            self.vm_state
+                .stack
+                .iter()
+                .enumerate()
+                .map(|(i, f)| format!("[{i}] {f}"))
+                .collect::<Vec<_>>()
+                .join("\n"),
+        );
```

Comment:
This change results in 

```
error[E0308]: mismatched types
   --> src/lib.rs:299:38
    |
299 |               |s, (i, f)| s.push_str(format!("[{i}] {f}\n"))
    |                                      ^^^^^^^^^^^^^^^^^^^^^^ expected `&str`, found struct `String`
    |
    = note: this error originates in the macro `format` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0308]: mismatched types
   --> src/lib.rs:299:27
    |
299 |               |s, (i, f)| s.push_str(format!("[{i}] {f}\n"))
    |                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `String`, found `()`
    |
note: method `push_str` modifies its receiver in-place
   --> src/lib.rs:299:29
    |
299 |               |s, (i, f)| s.push_str(format!("[{i}] {f}\n"))
    |                           - ^^^^^^^^ this call modifies `s` in-place
    |                           |
    |                           you probably want to use this value after calling the method...
    = note: ...instead of the `()` output of method `push_str`
    ```

Reply:
oh, sorry, I didn't test the code, have you tried something like this?

```rust
|s, (i, f)| {
    s.push_str(&format!("[{i}] {f}\n"));
    s
}
```